src/components/results/shortcircuit/shortcircuit-analysis-result-table.tsx

```diff
@@ -180,31 +227,19 @@ const ShortCircuitAnalysisResult = ({ result }) => {
     const renderResult = () => {
         const rows = flattenResult(result);
 
-        return (
-            result &&
-            shortCircuitNotif && (
-                <CustomAGGrid
-                    rowData={rows}
-                    columnDefs={columns}
-                    getRowStyle={getRowStyle}
-                    onGridReady={onGridReady}
-                    enableCellTextSelection={true}
-                    postSortRows={handlePostSortRows}
-                    defaultColDef={defaultColDef}
-                />
-            )
-        );
-    };
-
-    return renderResult();
-};
+        const aggridProps: AgGridReactProps = {
```

Comment:
why create an object with the props ? Is it just to get the type check ? Then I think we shouldn't do it, type checks should not force you to write weird code

Reply:
it was actually a typecheck problem due to the usage of a JS file and useRef
since I could not find a quick satisfying solution, I migrated the component to typescript, which fixed the problem